### DISCUSSION TITLE
CS50 - Finance : Codespace is timing out today

### DISCUSSION THREAD

**Discussion by matthewsmithwarren** [community/community#56064] [Codespaces] [answered]
### Select Topic Area

Question

### Body

I'm in the final 2 weeks of Harvard's CS50 course. Working on Finance. GitHub Codespace has been working well most of the time, but today things seem to be out of service (getting a 504) although there's no mention of an outage on GitHub's status page.
![Screenshot 2023-05-21 at 3 52 28 PM](https://github.com/community/community/assets/14283993/326e3059-d586-40dd-bc48-28bdc06afef7)

I'm currently working with Flask/Python API application but checked a previous project that used http-server and both failed with the 504 when trying to connect with forwarded port to test the application. Gateway Timeout - GitHub

Detail in case they're helpful: Request ID: fc7e42c3f8a4d79d82149b873f57dbd2

**Comment by Mikens101** [comment]
I'm having the same issue. Any solution yet?

**Reply by matthewsmithwarren** [reply]
No solution yet, but comforting to see that others are seeing the same. It means I can skip trying to troubleshoot something I did to break it.

**Reply by matthewsmithwarren** [reply, marked as answer]
It's back up!

**Reply by matthewsmithwarren** [reply]
The problem I had was resolved but I was not directly involved. Shortly after I created this post, someone (CS50 staff or GitHub) fixed it. Your error looks different so you may want to start a new thread. I searched in Discord (Finance & IDE) channels to first see if others were experiencing the same. Then posted in those channels with details (eg. your "DNS probe .." above).

**Comment by matthewsmithwarren** [comment]
Hi Michael,

It did resolve. And I just submitted Finance. Curious how things turned out
for you.


*Matt*

**Reply by Mikens101** [reply]
Been working well ever since you posted! Thank you for the heads up when it was fixed!

**Comment by Ankitchaudhary810** [comment]
hey can you help me i am not able to open [https://cs50.harvard.edu/x/](url) i getting this error on my brower he connection has timed out

An error occurred during a connection to cs50.harvard.edu.

    The site could be temporarily unavailable or too busy. Try again in a few moments.
    If you are unable to load any pages, check your computer’s network connection.
    If your computer or network is protected by a firewall or proxy, make sure that Firefox is permitted to access the web. 
after changing the router several times.

**Comment by matthewsmithwarren** [comment]
Hi Ankit, I have completed the course and am no longer using the codespace.
I’m not much help. Recommend you start a new thread in Discord and the ping
the faculty.

On Mon, Jul 31, 2023 at 11:51 AM Ankit ***@***.***> wrote:

> hey can you help me i am not able to open https://cs50.harvard.edu/x/
> <http://url> i getting this error on my brower he connection has timed out
>
> An error occurred during a connection to cs50.harvard.edu.
>
> The site could be temporarily unavailable or too busy. Try again in a few moments.
> If you are unable to load any pages, check your computer’s network connection.
> If your computer or network is protected by a firewall or proxy, make sure that Firefox is permitted to access the web.
>
> after changing the router several times.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/community/discussions/56064#discussioncomment-6596778>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADM7JWPZMUJWJ6BXWOOSMM3XS7H7HANCNFSM6AAAAAAYJS5SYQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
*Matt*
Matthew Smith Warren <https://www.linkedin.com/in/matthewsmithwarren/>
*DevSecOps and Cloud **Software Solutions*
mobile: 216.401.6813
email: ***@***.***
schedule: Calendly
<https://calendly.com/matthewsmithwarren/25-minute-meeting>
<https://www.credly.com/badges/3e4e4cb9-baa7-41e2-b1f1-11736ef1f9af/public_url>

**Comment by CatherineXJJin** [comment]
I’m working on cs50 finance and this issue has come up again, or at least I think it’s something similar. I get 504 errors whenever I try to use flask run, even when I try it with a different flask problem (the birthdays one from week 8). It was working perfectly last night and the only change I made this morning was adding a table to the index. I can get flask run to work briefly by restarting my computer but before long it goes back to throwing 504 errors

**Comment by matthewsmithwarren** [comment]
Hi Catherine, I’m no longer in the Codespace regularly. You may want start
a new thread for visibility with the administrators.
